File: FFCG.G4.GameOfLife/FFCG.G4.GameOfLife/Cell.cs
using System.Collections.Generic;
using System.Linq;
using FFCG.G4.GameOfLife.Rules;

namespace FFCG.G4.GameOfLife
{
    public class Cell
    {
        private readonly IEnumerable<ILifeRule> _lifeRules;

        public Cell(IEnumerable<ILifeRule> lifeRules, bool shouldStartOutAlive = false)
        {
            _lifeRules = lifeRules;
            IsAlive = shouldStartOutAlive;
        }

        public bool IsAlive { get; set; }

        public Cell Split(int amountOfNeighbors)
        {
            var shouldLive = _lifeRules.Where(x => x.ShouldHandle(IsAlive)).All(x => x.ShouldLive(amountOfNeighbors));
            return new Cell(_lifeRules, shouldLive);
        }
    }
}

File: FFCG.G7.GameOfLife/src/Cell.ts
import { ILifeRule } from "./rules/ILifeRule";

export class Cell {
    lifeRules: Array<ILifeRule>
    isAlive: boolean

    constructor(lifeRules: Array<ILifeRule>, isAlive = false) {
        this.lifeRules = lifeRules
        this.isAlive = isAlive
    }

    split(numberOfNeighbors): Cell {
        const shouldLive = this.lifeRules.filter(r => r.canHandle(this.isAlive)).every(r => r.shouldLive(numberOfNeighbors))
        return new Cell(this.lifeRules, shouldLive)
    }
}